icd/api/pipeline_compiler.cpp

```diff
@@ -1050,26 +1050,79 @@ void PipelineCompiler::GetPipelineCreationInfoNext(
 
 // =====================================================================================================================
 VkResult PipelineCompiler::SetPipelineCreationFeedbackInfo(
-    const VkPipelineCreationFeedbackCreateInfoEXT* pPipelineCreationFeadbackCreateInfo,
-    const PipelineCreationFeedback*                pPipelineFeedback)
-{
-    if (pPipelineCreationFeadbackCreateInfo != nullptr)
-    {
-        if (pPipelineFeedback->feedbackValid)
-        {
-            pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
-                VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+    const VkPipelineCreationFeedbackCreateInfoEXT*
+        pPipelineCreationFeadbackCreateInfo,
+    const VkGraphicsPipelineCreateInfo* pCreateInfo,
+    const PipelineCreationFeedback* pPipelineFeedback,
+    const PipelineCreationFeedback* pStageFeedback) {
+  if (pPipelineCreationFeadbackCreateInfo != nullptr) {
+    if (pPipelineFeedback->feedbackValid) {
+      pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
+          VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+    }
+    if (pPipelineFeedback->hitApplicationCache) {
+      pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
+          VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
+    }
+    pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->duration =
+        pPipelineFeedback->duration;
+    if (pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks !=
+            nullptr &&
+        pStageFeedback != nullptr) {
+      // Compute pipeline does have VkGraphicsPipelineCreateInfo.
+      if (pCreateInfo == nullptr) {
+        pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks[0]
+            .flags |= VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT_EXT;
+        if (pStageFeedback[0].hitApplicationCache) {
+          pPipelineCreationFeadbackCreateInfo
+              ->pPipelineStageCreationFeedbacks[0]
+              .flags |=
+              VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
         }
-        if (pPipelineFeedback->hitApplicationCache)
-        {
-            pPipelineCreationFeadbackCreateInfo->pPipelineCreationFeedback->flags |=
-                VK_PIPELINE_CREATION_FEEDBACK_APPLICATION_PIPELINE_CACHE_HIT_BIT_EXT;
+        pPipelineCreationFeadbackCreateInfo->pPipelineStageCreationFeedbacks[0]
+            .duration = pStageFeedback[0].duration;
+        return VK_SUCCESS;
```

Comment:
The PAL coding style doesn't use early returns and typically places a result value at the top of the function body and updates it throughout the code

Reply:
done